File: wagerproof-ios-native/Wagerproof/Features/Onboarding/OnboardingTheme.swift
import SwiftUI
import WagerproofDesign
import WagerproofStores

/// Maps onboarding selections to the accent color that tints the reactive
/// pixel background, the CTA pill, and per-page highlights. The default is
/// the brand green so the entry state is visually continuous with the login
/// screen's `AuthGateBackground` (same `PixelWaveBackground`, same tint).
enum OnboardingTheme {
    static func accent(for type: OnboardingStore.BettorType?) -> Color {
        switch type {
        case .none, .casual:
            return .appPrimary
        case .serious:
            return .appAccentBlue
        case .professional:
            return .appAccentPurple
        }
    }

    /// Archetype rows carry their own hex accent (`PresetArchetypeRow.color`).
    static func archetypeAccent(hex: String?) -> Color? {
        guard let hex else { return nil }
        return Color(hexString: hex)
    }

    /// Slightly white-lifted accent used while the generation cinematic runs
    /// — reads as "energized" without touching the glyph field's grid params.
    static func generationBoost(_ accent: Color) -> Color {
        accent.mix(with: .white, by: 0.15)
    }
}

File: wagerproof-android-native/app/src/main/java/com/wagerproof/app/features/onboarding/OnboardingTheme.kt
package com.wagerproof.app.features.onboarding

import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.lerp
import com.wagerproof.app.features.agents.colorFromHexString
import com.wagerproof.core.design.tokens.AppColors
import com.wagerproof.core.stores.OnboardingStore

/**
 * Port of iOS `OnboardingTheme.swift`. Maps onboarding selections to the accent
 * color that tints the reactive pixelwave background, the CTA pill, and per-page
 * highlights. The default is brand green so the entry state stays visually
 * continuous with the login screen's pixelwave (same tint).
 */
object OnboardingTheme {

    /** Bettor-type accent — null/casual green, serious blue, professional purple. */
    fun accent(type: OnboardingStore.BettorType?): Color = when (type) {
        null, OnboardingStore.BettorType.Casual -> AppColors.appPrimary
        OnboardingStore.BettorType.Serious -> AppColors.appAccentBlue
        OnboardingStore.BettorType.Professional -> AppColors.appAccentPurple
    }

    /** Archetype rows carry their own hex accent (`PresetArchetypeRow.color`). */
    fun archetypeAccent(hex: String?): Color? = hex?.let { colorFromHexString(it) }

    /**
     * Slightly white-lifted accent used while the generation cinematic runs —
     * reads as "energized" without touching the glyph field's grid params.
     * iOS `accent.mix(with: .white, by: 0.15)`.
     */
    fun generationBoost(accent: Color): Color = lerp(accent, Color.White, 0.15f)
}
